build.rs

```diff
@@ -0,0 +1,108 @@
+// following https://doc.rust-lang.org/cargo/reference/build-scripts.html
```

Comment:
i'm not sure what the implications of having this `build.rs` is for library consumers... it seems to me that it'll always be invoked which would cause some pretty weird behavior if the higher level project defined the right MCU name.

maybe it'd be better to have an `examples` sub-crate as proposed in the discussion around optional dev-dependencies so they can be split? alternately it may be possible to feature-gate the build script so it only runs when building _these_ examples (as distinct from, examples for a package that depends on this 😂)

Reply:
It might be troublesome having magic making examples work. That would cause problems for anyone playing around and giving a new name to a small change. I think there should be better options.

Subcrates seem to complicate submission to crates.io. From `The Cargo Book` (https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html):
```
However, crates that use dependencies specified with only a path are not 
permitted on crates.io. If we wanted to publish our hello_world crate, we would 
need to publish a version of hello_utils to crates.io and specify its version in 
the dependencies line as well: ...
```

Reply:
This is completely entwined with the next conversation on `.cargo/config`. I'll comment there and close this.